tests/Doctrine/Tests/ORM/UnitOfWorkTest.php

```diff
@@ -705,6 +709,50 @@ public function testCanInstantiateInternalPhpClassSubclassFromUnserializedMetada
 
         self::assertInstanceOf(MyArrayObjectEntity::class, $this->unitOfWork->newInstance($classMetadata));
     }
+
+    public function testCallCommitInsideCommit() : void
+    {
+        $listener = new class()
+        {
+            public function preFlush(PreFlushEventArgs $eventArgs)
+            {
+                $eventArgs->getEntityManager()->getUnitOfWork()->commit();
+            }
+        };
+
+        $this->expectException(CommitInsideCommit::class);
+        $this->eventManager->addEventListener(Events::preFlush, $listener);
```

Comment:
It's a best practice to move the `expectException` calls right before the calls that throw the exception.

```suggestion
        $this->eventManager->addEventListener(Events::preFlush, $listener);
        $this->expectException(CommitInsideCommit::class);
```